### ISSUE TITLE
WP CLI is Broken by XPress

### ISSUE THREAD

**Issue by ichpen** [Audentio/xpress-issues#597]
Reopening this as Audentio/xf2addon-issues#2084 was closed with an irrelevant explanation.

Let me clarify the issue again.

1. Express installed with a WP install in a subdirectory
2. When I attempt to use WP CLI in that subdir it breaks with the below output:

> "Warning: file_exists(): Unable to find the wrapper "phar" - did you forget to enable it when you configured PHP? in /root/.wp-cli/packages/vendor/composer/ClassLoader.php on line 394
> An unexpected error occurred. Please try again later."
> 

3. When I disable/remove the Express plugin WP CLI works perfectly in the same subdirectory

This has been recreated with every 7.x.x version of PHP.

I don't believe this is a third party plugin issue as it's directly influenced by the presence of XPress. And I use WP CLI a fair bit in my install.

Xpress does appear to be working otherwise but it breaks WP CLI.

**Comment by LukasWieditz** [comment]
We've noticed a handful of plugins that seemingly disable the phar wrapper for one reason or another. It's required by composer, which is in turn a requirement for XenForo (and XPress), so there's not a whole lot we can do to counteract that.

**Comment by teamracing** [comment]
I give up

**Comment by LukasWieditz** [comment]
What do you expect us to do? Magically make the other add-on not disable a required extension?

**Comment by teamracing** [comment]
> What do you expect us to do? Magically make the other add-on not disable a required extension?

It's not another addon it's the Themehouse Xpress addon causing this. When I disable(delete) your addon, CLI magically works. This is what I've been explaining to you across two bug reports. It's not a handful of addons it is your addon by the looks of it.

Guess what, other people over at centminmod are experiencing identical issues with xpress (see here): https://community.centminmod.com/threads/some-wordpress-issues.19068/

I'm not being difficult here, it's just that your addon is causing WP CLI to break, I cannot put it in simpler terms. If it's another issue please let us know so we can take corrective steps. Right now the alternative is Xpress and no CLI or no CLI with Xpress.

Help some of us please or at least tell us where to investigate if you have some ideas. Maybe it's a centmin problem... I can only tell you what fixes it today.

**Comment by LukasWieditz** [comment]
Yes, XPress is throwing an error, because a different add-on has disabled the phar wrapper, as stated by your error report above and as I've said in my closing comment. As long as a different add-on keeps disabling the wrapper, XPress will continue to throw that error. It isn't even XPress throwing that error technically, XenForo would as well, even if XPress itself wouldn't use composer.
